sql-streaming-sqs/src/main/scala/org/apache/spark/sql/streaming/sqs/SqsClient.scala

```diff
@@ -131,13 +131,24 @@ class SqsClient(sourceOptions: SqsSourceOptions,
     }
   }
 
+  private def extractS3Message(parsedBody: JValue): JValue = {
+    implicit val formats = DefaultFormats
+    sourceOptions.messageWrapper match {
+      case sourceOptions.S3MessageWrapper.None => parsedBody
+      case sourceOptions.S3MessageWrapper.SNS => parse((parsedBody \ "Message").extract[String])
```

Comment:
Done. Added small section which describes differences.

Reply:
I see. There's a retry on failure mechanism for error in fetching messages from SQS. Maybe we should have an option for retry on failure in case of an error in parsing messages too. Of course, it's not related to this and can be done as a part of a separate PR.

Reply:
Right now if we fail to parse SQS we just remove these messages from SQS. I think what would make sense is try to refetch them again. And agree, this should be done in another PR. I can raise a ticket and make these changes. Thoughts/Objections?

Reply:
In my use case, I've seen some extra messages in the SQS queue in the following scenarios:
- When S3 bucket is configured to SQS queue during pipeline setup
- some configuration is changed 
- some test messages. 

There are not `ObjectCreate` or `ObjectRemoved ` messages and so they are not relevant for spark consumer application so it seemed reasonable to ignore and delete them. 

But if the spark consumer receives too many unparsable messages, it means there might be an issue with the pipeline set up. so according to me, it makes sense to throw an error and abort the pipeline with the help of a finite max retries mechanism.

Refetching and trying to parse the same message again may help if a legitimate message gets corrupted while fetching it from SQS. Can you list the scenarios where refetching and trying to parse the same message will help?

Feel free to raise a ticket and start a PR. We can discuss this further over there.

Reply:
Thanks for the details! I will raise a ticket shortly